pages/api/admin/CheckPreviousVisit.js

```diff
@@ -0,0 +1,182 @@
+import { validateFunc } from "../validate";
+import { google } from "googleapis";
+
+import {service_info} from "../../../utils/decrypt.js";
+import firebase from '../../../firebase/clientApp'
+
+/*
+ * /api/admin/CheckPreviousVisit
+ * req.body = { string calID, boolean isGrad }
+ */
+
+function requireParams(body) {
+  // makes sure that the input is in the right format
+  // returns false and an error if not a good input
+  if (body.calID && body.isGrad != null) return true;
+  return false;
+}
+
+//returns the start of the week given a day
+function determineStartOfWeek(currDay) {
+  let startOfWeek = new Date(currDay.toLocaleDateString());
+  const millisecondsPerDay = 86400000;
+  const milSecondsUntilStart = millisecondsPerDay * currDay.getDay();
+  startOfWeek.setTime(startOfWeek.getTime() - milSecondsUntilStart);
+  return startOfWeek.getTime();
+}
+
+//converts from 2022-07-23T20:35:41.935Z to 7/23/2022 12:15:52
+function formatTime(timeToConvert) {
+const formattedHours = timeToConvert.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit', second:'2-digit', timeZone: 'America/Los_Angeles'})
+const formattedTime = timeToConvert.toLocaleDateString('en-US', {timeZone: 'America/Los_Angeles'}) + " " + formattedHours;
+    return formattedTime;
+}
+
+//converts from 2022-07-23T20:35:41.935Z to Mon Jul 23 2022 at 08:35 PM
+function formatTimeForVisits(timeToConvert) {
+  const formattedHours = timeToConvert.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit'})
+  const formattedTime = timeToConvert.toDateString('en-US', {timeZone: 'America/Los_Angeles'}) + " at " + formattedHours;
+  return formattedTime;
+  }
+
+//get the number of rows for the Check Out sheet
+function getNumRowsForCheckIn(properties) {
+  for (var sheet of properties["data"]["sheets"]) {
+    if (sheet.properties.title == "Check Out Form") {
+      return sheet.properties.gridProperties.rowCount;
+    }
+  }
+}
```

Comment:
```suggestion
function getNumRowsForCheckIn(properties, sheetName) {
  for (var sheet of properties["data"]["sheets"]) {
    if (sheet.properties.title == sheetName) {
      return sheet.properties.gridProperties.rowCount;
    }
  }
}
```
should use the firebase sheetName instead of hardcoding, just in case we rename the sheet i think? need to pass in sheetName below